elementary/appcenter
# Simplify view handling

This PR simplifies how communication between the MainWindow and it's views happen. Previously the code logic for handling how information would be displayed in the header bar was in several places, and it was not exactly clear what it does in order to show / hide / change widgets. This branch attempts separating this code logic so that it is in it's respective views. In particular the `view_opened` and `view_return` methods don't contain any view-specific code now. they now only take the parameters from each widget about what should it display and do only that. 

This branch should indirectly fix #369.

To test:
There should be no different behaviour than what is now currently in master. What should be tested is if everything in the header bar is updated and works properly when you enter a new view.
